.github/workflows/preview_build.yml

```diff
@@ -0,0 +1,132 @@
+name: Preview build
+
+on:
+  repository_dispatch:
+    types:
+    - Preview*
+
+jobs:
+  preview:
+    name: Build preview
+
+    runs-on: ubuntu-20.04
+    container:
+      # If you change this version, change the number in the cache step too.
+      image: emscripten/emsdk:2.0.10
+      # uid=1001(runner) gid=121(docker)
+      options: -u 1001:121
+
+    steps:
+    - name: Update deployment status to in progress
+      uses: octokit/request-action@v2.x
+      with:
+        route: POST /repos/{owner}/{repo}/deployments/{deployment_id}/statuses
+        mediaType: |
+          previews:
+          - ant-man
+          - flash
+        owner: ${{ github.event.repository.owner.login }}
+        repo: ${{ github.event.repository.name }}
+        deployment_id: ${{ github.event.client_payload.deployment_id }}
+        state: in_progress
+      env:
+        GITHUB_TOKEN: ${{ secrets.PREVIEW_GITHUB_TOKEN }}
+
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        ref: ${{ github.event.client_payload.sha }}
+
+    - name: Name branch
+      run: |
+        name=$(echo "${{ github.event.client_payload.folder }}")
+        git checkout -b ${name}
+
+    - name: Setup cache
+      uses: actions/cache@v2
+      with:
+        path: /emsdk/upstream/emscripten/cache
+        key: 2.0.10-${{ runner.os }}
+
+    - name: Build (host tools)
+      run: |
+        mkdir build-host
+        cd build-host
+
+        echo "::group::CMake"
+        cmake .. -DOPTION_TOOLS_ONLY=ON
+        echo "::endgroup::"
+
+        echo "::group::Build"
+        echo "Running on $(nproc) cores"
+        make -j$(nproc) tools
+        echo "::endgroup::"
+
+    - name: Install GCC problem matcher
+      uses: ammaraskar/gcc-problem-matcher@master
+
+    - name: Build
+      run: |
+        mkdir build
+        cd build
+
+        echo "::group::CMake"
+        emcmake cmake .. \
+          -DHOST_BINARY_DIR=../build-host \
+          -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+          -DOPTION_USE_ASSERTS=OFF \
```

Comment:
The documentation isn't particularly clear, but I think asserts are already disabled in RelWithDebInfo BUILD_TYPES (certainly is with Release, why not use that?)